pkg/minishift/hostfolder/hostfolder.go

```diff
@@ -0,0 +1,428 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hostfolder
+
```

Comment:
At the moment I have mount specific code for CIFS in `mountCifsHostFolder`. All the basic commands `Mount`, 'Umount', `List` have been written to prevent using any implementation specific code. For the current `Add` this is not done. It is now checked against `HostFolder.Type`. HostFolder has actually been extracted early with general `Options[]` for the reason of re-use among implementation. This all is in my opinion a good candidate for an interface. Just when I worked on it, I would have introduced to premature.